algebra/src/fields/models/cubic_extension.rs

```diff
@@ -0,0 +1,501 @@
+use std::{
+    cmp::{Ord, Ordering, PartialOrd},
+    fmt,
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    io::{Read, Result as IoResult, Write},
+};
+use rand::{
+    distributions::{Distribution, Standard},
+    Rng,
+};
+
+use crate::{
+    bytes::{FromBytes, ToBytes},
+    bits::{FromBits, ToBits},
+    fields::{Field, FpParameters, PrimeField},
+    UniformRand, Error,
+};
+
+
+/// Model for cubic extension field of a prime field F=BasePrimeField
```

Comment:
wrong comment here, should be cubic extension of F=BaseField (which again is an extension of BasePrimeField).